lib/collection/src/operations/consistency_params.rs

```diff
@@ -53,23 +32,150 @@ impl Default for ReadConsistency {
     }
 }
 
-impl<'de> serde::Deserialize<'de> for ReadConsistency {
-    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        let consistency = ReadConsistency::deserialize(deserializer)?;
-        if let ReadConsistency::Factor(factor) = &consistency {
-            if *factor == 0 {
-                return Err(D::Error::custom(ValidationError));
-            }
+impl ReadConsistency {
+    pub fn try_from_optional(
+        consistency: Option<ReadConsistencyGrpc>,
+    ) -> Result<Option<Self>, tonic::Status> {
+        consistency.map(TryFrom::try_from).transpose()
+    }
+}
+
+impl TryFrom<Option<ReadConsistencyGrpc>> for ReadConsistency {
+    type Error = tonic::Status;
+
+    fn try_from(consistency: Option<ReadConsistencyGrpc>) -> Result<Self, Self::Error> {
+        match consistency {
+            Some(consistency) => consistency.try_into(),
+            None => Ok(Self::Factor(1)),
         }
+    }
+}
+
+impl TryFrom<ReadConsistencyGrpc> for ReadConsistency {
+    type Error = tonic::Status;
+
+    fn try_from(consistency: ReadConsistencyGrpc) -> Result<Self, Self::Error> {
+        let value = consistency.value.ok_or_else(|| {
+            tonic::Status::invalid_argument(
+                "invalid read consistency message: `ReadConsistency::value` field is `None`",
+            )
+        })?;
+
+        let consistency = match value {
+            read_consistency::Value::Factor(factor) => Self::Factor(factor.try_into().unwrap()),
```

Comment:
is it possible to remove the `unwrap` with `try_into()?` here?

Reply:
Sure. I'll fix it.

Reply:
great, there is another one in the diff I believe :)

Reply:
It's in the `From` implementation, so a bit more annoying to fix.